fs/unpacker.go

```diff
@@ -49,26 +50,63 @@ func NewLayerArchive() Archive {
 	return &layerArchive{}
 }
 
-func (la *layerArchive) Apply(ctx context.Context, root string, r io.Reader, opts ...archive.ApplyOpt) (int64, error) {
+// wrapReader uses a TeeReader to feed content into verifier (compressed or uncompressed).
+// If verification is disabled, this is a no-op.
+func wrapReader(r io.Reader, verifier *layerVerifier, compressed bool) io.Reader {
```

Comment:
Combined with the suggested change to make `layerVerifier` always non-nil, can we replace this with:
```
verifier.WrapCompressedReader(r)
```
and 
```
verifier.WrapUncompressedReader(r)
```